docs/guides/security-guide.md

```diff
@@ -1,141 +1,95 @@
 
-```text
--ca_cert_file=[CA certificate file]
--server_cert_file=[Server certificate file]
--server_key_file=[Server private key file]
-```
+#### 2.1.2 Requirements for gRPC clients
 
-For example:
+Under default conditions, the gRPC clients will require the TLS certificates
+to establish communication with infrap4d. The clients will need
+to use the same ca.crt file and the client.key and client.crt files signed
+by the ca.crt (can copy the generated files from the server if client is not
+on the same system as server).
 
-```bash
-# Files present in /tmp/certs/ directory
-$IPDK_RECIPE/install/sbin/infrap4d \
-    -ca_cert_file=/tmp/certs/ca.crt \
-    -server_cert_file=/tmp/certs/stratum.crt \
-    -server_key_file=/tmp/certs/stratum.key
-```
+- P4RT client
 
-### Client certificate verification
+    The P4Runtime Control client will default to communicating in secure mode
+using port 9559. If certificates are not available, the P4RT client will attempt
+a connection using insecure client credentials as a fallback mechanism.
+Note that the communication will fail if infrap4d runs in secure mode.
+Both server and client must specify insecure mode for this to work.
 
-infrap4d requires connecting gRPC clients to send a valid certificate
-that can be verified. A flag is available to tune the level of security required.
-The available values are:
+- gNMI client
 
-```text
-NO_REQUEST_CLIENT_CERT
-REQUEST_CLIENT_CERT_NO_VERIFY
-REQUEST_CLIENT_CERT_AND_VERIFY
-REQUIRE_CLIENT_CERT_NO_VERIFY
-REQUIRE_CLIENT_CERT_AND_VERIFY (default)
-```
+    gnmi-ctl (the gNMI client for DPDK) and sgnmi_cli (the secure gNMI client)
+issue requests to port 9339.
 
-More info on these values can be found on [this gRPC library documentation page](https://grpc.github.io/grpc/cpp/grpc__security__constants_8h.html#a29ffe63a8bb3b4945ecab42d82758f09).
+#### 2.1.3 Generate and install TLS certificates
 
-### Running in insecure mode
+See [Install TLS Certificates](https://github.com/ipdk-io/networking-recipe/blob/main/docs/guides/install-tls-certificates.md)
+for step by step guide to generate and install TLS certificates
 
-Ports can be opened in insecure mode if needed. This is controlled
-by a flag that needs to be enabled at runtime. Change the
-`grpc_open_insecure_mode` value to `true` to allow insecure communication
-Also, make sure `certs` directory is removed from the default location mentioned
-above.
+### 2.2 Running in insecure mode
 
 To launch infrap4d in insecure mode:
 
 ```bash
 $IPDK_RECIPE/install/sbin/infrap4d  -grpc_open_insecure_mode=true
 ```
 
-## gRPC clients
+To launch clients in insecure mode:
 
-Under default conditions, the gRPC clients will require the TLS certificates
-to establish communication with infrap4d. The clients will need
-to use the same ca.crt file and the client.key and client.crt files signed
-by the ca.crt (can copy the generated files from the server if client is not
-on the same system as server).
+For DPDK target:
```

Comment:
Does this need to be broken down for different targets? We can simplify this by simply saying that user needs to pass the `-grpc_use_insecure_mode=true` flag

Reply:
I specified different targets because we use sgnmi_cli for ES2K and gnmi-ctl for DPDK.